From 6424055431eb12477269206e22f048828ee4fd6d Mon Sep 17 00:00:00 2001
From: Donghwa Lee <dh09.lee@samsung.com>
Date: Wed, 7 Aug 2013 14:07:29 +0900
Subject: [PATCH 0521/1302] ARM: dts: remove repeated code and bug fix

This patch removes already defined dts code and include "exynos4412-slp_pq.dts".
And it is fixed gpio key interrupts.

Signed-off-by: Donghwa Lee <dh09.lee@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 arch/arm/boot/dts/exynos4412-redwood.dts | 661 +------------------------------
 1 file changed, 3 insertions(+), 658 deletions(-)

diff --git a/arch/arm/boot/dts/exynos4412-redwood.dts b/arch/arm/boot/dts/exynos4412-redwood.dts
index e3ef5fe..e6f50c9 100644
--- a/arch/arm/boot/dts/exynos4412-redwood.dts
+++ b/arch/arm/boot/dts/exynos4412-redwood.dts
@@ -12,71 +12,16 @@
  * published by the Free Software Foundation.
 */
 
-/dts-v1/;
-/include/ "exynos4412.dtsi"
+/include/ "exynos4412-slp_pq.dts"
 
 / {
 	model = "Samsung Redwood based on Exynos4412";
 	compatible = "samsung,redwood", "samsung,exynos4412";
 
-	memory {
-		reg =  <0x40000000 0x10000000
-			0x50000000 0x10000000
-			0x60000000 0x10000000
-			0x70000000 0x10000000>;
-
-		reserved-memory {
-			#address-cells = <1>;
-			#size-cells = <1>;
-
-			contig_mem: region@71000000 {
-				compatible = "linux,contiguous-memory-region";
-				reg = <0x71000000 0x0c000000>;
-				linux,default-contiguous-region;
-			};
-
-			mfc_l_mem: mfc_l_region@43000000 {
-				compatible = "linux,contiguous-memory-region", "reserved-memory-region";
-				reg = <0x43000000 0x1000000>;
-			};
-
-			mfc_r_mem: mfc_r_region@51000000 {
-				compatible = "linux,contiguous-memory-region", "reserved-memory-region";
-				reg = <0x51000000 0x1000000>;
-			};
-		};
-	};
-
 	chosen {
 		bootargs = "console=ttySAC2,115200N8 root=/dev/mmcblk0p5 rootwait earlyprintk panic=5";
 	};
 
-	firmware@0204F000 {
-		compatible = "samsung,secure-firmware";
-		reg = <0x0204F000 0x1000>;
-	};
-
-	fixed-rate-clocks {
-		xxti {
-			compatible = "samsung,clock-xxti", "fixed-clock";
-			clock-frequency = <0>;
-		};
-
-		xusbxti {
-			compatible = "samsung,clock-xusbxti", "fixed-clock";
-			clock-frequency = <24000000>;
-		};
-	};
-
-	vemmc_reg: voltage-regulator@0 {
-	        compatible = "regulator-fixed";
-		regulator-name = "VMEM_VDD_2.8V";
-		regulator-min-microvolt = <2800000>;
-		regulator-max-microvolt = <2800000>;
-		gpio = <&gpk0 2 0>;
-		enable-active-high;
-	};
-
 	sdhci_emmc: sdhci@12510000 {
 		bus-width = <8>;
 		non-removable;
@@ -86,29 +31,13 @@
 		status = "okay";
 	};
 
-	serial@13800000 {
-		status = "okay";
-	};
-
-	serial@13810000 {
-		status = "okay";
-	};
-
-	serial@13820000 {
-		status = "okay";
-	};
-
-	serial@13830000 {
-		status = "okay";
-	};
-
 	gpio-keys@0 {
 		compatible = "gpio-keys";
 
 		key@114 {
 			interrupt-parent = <&gpj1>;
 			interrupts = <2 0>;
-			gpios = <&gpj1 2 1>;
+			gpios = <&gpx3 3 1>;
 			linux,code = <114>;
 			label = "volume down";
 			debounce-interval = <10>;
@@ -117,7 +46,7 @@
 		key@115 {
 			interrupt-parent = <&gpj1>;
 			interrupts = <1 0>;
-			gpios = <&gpj1 1 1>;
+			gpios = <&gpx2 2 1>;
 			linux,code = <115>;
 			label = "volume up";
 			debounce-interval = <10>;
@@ -144,431 +73,6 @@
 		};
 	};
 
-	i2c_0: i2c@13860000 {
-		samsung,i2c-sda-delay = <100>;
-		samsung,i2c-slave-addr = <0x10>;
-		samsung,i2c-max-bus-freq = <400000>;
-		pinctrl-0 = <&i2c0_bus>;
-		pinctrl-names = "default";
-		status = "okay";
-
-		s5c73m3@3c {
-			compatible = "samsung,s5c73m3";
-			reg = <0x3c>;
-			gpios = <&gpm0 1 1>, /* ISP_STANDBY */
-				<&gpf1 3 1>; /* ISP_RESET */
-			vdd-int-supply = <&buck9_reg>;
-			vddio-cis-supply = <&ldo9_reg>;
-			vdda-supply = <&ldo17_reg>;
-			vddio-host-supply = <&ldo18_reg>;
-			vdd-af-supply = <&cam_af_reg>;
-			vdd-reg-supply = <&cam_io_reg>;
-			clock-frequency = <24000000>;
-
-			port {
-				s5c73m3_ep: endpoint {
-					remote-endpoint = <&csis0_ep>;
-				};
-			};
-		};
-	};
-
-	i2c@13870000 {
-		samsung,i2c-sda-delay = <100>;
-		samsung,i2c-slave-addr = <0x10>;
-		samsung,i2c-max-bus-freq = <400000>;
-		pinctrl-0 = <&i2c1_bus>;
-		pinctrl-names = "default";
-		status = "okay";
-
-		lsm330dlc_gyro@6b {
-			compatible = "st,lsm330dlc-gyro";
-			reg = <0x6b>;
-			irq-map-policy = <1>;
-			interrupt-controller;
-			#interrups-cells = <2>;
-            		interrupt-parent = <&lsm330dlc_gyro_map>;
-            		interrupts= <1 0>;
-
-			lsm330dlc_gyro_map: lsm330dlc-gyro-map {
-                		compatible = "samsung,lsm330dlc-gyro-map";
-                		#interrupt-cells = <2>;
-                		#address-cells = <0>;
-                		#size-cells = <0>;
-                		interrupt-map = <0x1 0 &gpf0 3 0>;
-            		};
-		};
-
-		lsm330dlc_accel@19 {
-			compatible = "st,lsm330dlc-accel";
-			reg = <0x19>;
-			irq-map-policy = <2>;
-			interrupt-controller;
-            		#interrups-cells = <2>;
-            		interrupt-parent = <&lsm330dlc_accel_map>;
-            		interrupts= <1 0>;
-
-			lsm330dlc_accel_map: lsm330dlc-accel-map {
-                		compatible = "samsung,lsm330dlc-accel-map";
-                		#interrupt-cells = <2>;
-                		#address-cells = <0>;
-                		#size-cells = <0>;
-                		interrupt-map = <0x1 0 &gpx0 0 0>;
-            		};
-		};
-	};
-
-	i2c@13890000 {
-		samsung,i2c-sda-delay = <100>;
-		samsung,i2c-slave-addr = <0x10>;
-		samsung,i2c-max-bus-freq = <400000>;
-		pinctrl-0 = <&i2c3_bus>;
-		pinctrl-names = "default";
-		status = "okay";
-
-		mms114-touchscreen@48 {
-			compatible = "melfas,mms114";
-			reg = <0x48>;
-			interrupt-parent = <&gpm2>;
-			interrupts = <3 2>;
-			x-size = <720>;
-			y-size = <1280>;
-			avdd-supply = <&ldo23_reg>;
-			vdd-supply = <&ldo24_reg>;
-		};
-	};
-
-	i2c@138D0000 {
-		samsung,i2c-sda-delay = <100>;
-		samsung,i2c-slave-addr = <0x10>;
-		samsung,i2c-max-bus-freq = <100000>;
-		pinctrl-0 = <&i2c7_bus>;
-		pinctrl-names = "default";
-		status = "okay";
-
-		max77686_pmic@09 {
-			compatible = "maxim,max77686";
-			interrupt-parent = <&gpx0>;
-			interrupts = <7 0>;
-			reg = <0x09>;
-
-			voltage-regulators {
-				ldo1_reg: ldo@1 {
-					regulator-compatible = "LDO1";
-					regulator-name = "VALIVE_1.0V_AP";
-					regulator-min-microvolt = <1000000>;
-					regulator-max-microvolt = <1000000>;
-					regulator-always-on;
-				};
-
-				ldo2_reg: ldo@2 {
-					regulator-compatible = "LDO2";
-					regulator-name = "VM1M2_1.2V_AP";
-					regulator-min-microvolt = <1200000>;
-					regulator-max-microvolt = <1200000>;
-					regulator-always-on;
-				};
-
-				ldo3_reg: ldo@3 {
-					regulator-compatible = "LDO3";
-					regulator-name = "VCC_1.8V_AP";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-					regulator-always-on;
-				};
-
-				ldo4_reg: ldo@4 {
-					regulator-compatible = "LDO4";
-					regulator-name = "VCC_2.8V_AP";
-					regulator-min-microvolt = <2800000>;
-					regulator-max-microvolt = <2800000>;
-					regulator-always-on;
-				};
-
-				ldo5_reg: ldo@5 {
-					regulator-compatible = "LDO5";
-					regulator-name = "VCC_1.8V_IO";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-					regulator-always-on;
-				};
-
-				ldo6_reg: ldo@6 {
-					regulator-compatible = "LDO6";
-					regulator-name = "VMPLL_1.0V_AP";
-					regulator-min-microvolt = <1000000>;
-					regulator-max-microvolt = <1000000>;
-					regulator-always-on;
-				};
-
-				ldo7_reg: ldo@7 {
-					regulator-compatible = "LDO7";
-					regulator-name = "VPLL_1.0V_AP";
-					regulator-min-microvolt = <1000000>;
-					regulator-max-microvolt = <1000000>;
-					regulator-always-on;
-				};
-
-				ldo8_reg: ldo@8 {
-					regulator-compatible = "LDO8";
-					regulator-name = "VMIPI_1.0V";
-					regulator-min-microvolt = <1000000>;
-					regulator-max-microvolt = <1000000>;
-				};
-
-				ldo9_reg: ldo@9 {
-					regulator-compatible = "LDO9";
-					regulator-name = "CAM_ISP_MIPI_1.2V";
-					regulator-min-microvolt = <1200000>;
-					regulator-max-microvolt = <1200000>;
-				};
-
-				ldo10_reg: ldo@10 {
-					regulator-compatible = "LDO10";
-					regulator-name = "VMIPI_1.8V";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-				};
-
-				ldo11_reg: ldo@11 {
-					regulator-compatible = "LDO11";
-					regulator-name = "VABB1_1.8V";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-					regulator-always-on;
-				};
-
-				ldo12_reg: ldo@12 {
-					regulator-compatible = "LDO12";
-					regulator-name = "VUOTG_3.0V";
-					regulator-min-microvolt = <3000000>;
-					regulator-max-microvolt = <3000000>;
-				};
-
-				ldo13_reg: ldo@13 {
-					regulator-compatible = "LDO13";
-					regulator-name = "NFC_AVDD_1.8V";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-				};
-
-				ldo14_reg: ldo@14 {
-					regulator-compatible = "LDO14";
-					regulator-name = "VABB2_1.8V";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-					regulator-always-on;
-				};
-
-				ldo15_reg: ldo@15 {
-					regulator-compatible = "LDO15";
-					regulator-name = "VHSIC_1.0V";
-					regulator-min-microvolt = <1000000>;
-					regulator-max-microvolt = <1000000>;
-					regulator-always-on;
-				};
-
-				ldo16_reg: ldo@16 {
-					regulator-compatible = "LDO16";
-					regulator-name = "VHSIC_1.8V";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-					regulator-always-on;
-				};
-
-				ldo17_reg: ldo@17 {
-					regulator-compatible = "LDO17";
-					regulator-name = "CAM_SENSOR_CORE_1.2V";
-					regulator-min-microvolt = <1200000>;
-					regulator-max-microvolt = <1200000>;
-				};
-
-				ldo18_reg: ldo@18 {
-					regulator-compatible = "LDO18";
-					regulator-name = "CAM_ISP_SENSOR_1.8V";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-				};
-
-				ldo19_reg: ldo@19 {
-					regulator-compatible = "LDO19";
-					regulator-name = "VT_CAM_1.8V";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-				};
-
-				ldo20_reg: ldo@20 {
-					regulator-compatible = "LDO20";
-					regulator-name = "VDDQ_PRE_1.8V";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-				};
-
-				ldo21_reg: ldo@21 {
-					regulator-compatible = "LDO21";
-					regulator-name = "VTF_2.8V";
-					regulator-min-microvolt = <2800000>;
-					regulator-max-microvolt = <2800000>;
-				};
-
-				ldo22_reg: ldo@22 {
-					regulator-compatible = "LDO22";
-					regulator-name = "VMEM_VDD_2.8V";
-					regulator-min-microvolt = <2800000>;
-					regulator-max-microvolt = <2800000>;
-					regulator-always-on;
-				};
-
-				ldo23_reg: ldo@23 {
-					regulator-compatible = "LDO23";
-					regulator-name = "TSP_AVDD_3.3V";
-					regulator-min-microvolt = <3300000>;
-					regulator-max-microvolt = <3300000>;
-				};
-
-				ldo24_reg: ldo@24 {
-					regulator-compatible = "LDO24";
-					regulator-name = "VDD_1.8V_TSP";
-					regulator-min-microvolt = <1800000>;
-					regulator-max-microvolt = <1800000>;
-				};
-
-				ldo25_reg: ldo@25 {
-					regulator-compatible = "LDO25";
-					regulator-name = "LED_A_2.8V";
-					regulator-min-microvolt = <2800000>;
-					regulator-max-microvolt = <2800000>;
-				};
-
-				ldo26_reg: ldo@26 {
-					regulator-compatible = "LDO26";
-					regulator-name = "MOTOR_VCC_3.0V";
-					regulator-min-microvolt = <3000000>;
-					regulator-max-microvolt = <3000000>;
-				};
-
-				buck1_reg: buck@1 {
-					regulator-compatible = "BUCK1";
-					regulator-name = "vdd_mif";
-					regulator-min-microvolt = <850000>;
-					regulator-max-microvolt = <1100000>;
-					regulator-always-on;
-					regulator-boot-on;
-				};
-
-				buck2_reg: buck@2 {
-					regulator-compatible = "BUCK2";
-					regulator-name = "vdd_arm";
-					regulator-min-microvolt = <850000>;
-					regulator-max-microvolt = <1500000>;
-					regulator-always-on;
-					regulator-boot-on;
-				};
-
-				buck3_reg: buck@3 {
-					regulator-compatible = "BUCK3";
-					regulator-name = "vdd_int";
-					regulator-min-microvolt = <850000>;
-					regulator-max-microvolt = <1150000>;
-					regulator-always-on;
-					regulator-boot-on;
-				};
-
-				buck4_reg: buck@4 {
-					regulator-compatible = "BUCK4";
-					regulator-name = "vdd_g3d";
-					regulator-min-microvolt = <850000>;
-					regulator-max-microvolt = <1150000>;
-					regulator-boot-on;
-				};
-
-				buck5_reg: buck@5 {
-					regulator-compatible = "BUCK5";
-					regulator-name = "VMEM_1.2V_AP";
-					regulator-min-microvolt = <1200000>;
-					regulator-max-microvolt = <1200000>;
-					regulator-always-on;
-				};
-
-				buck6_reg: buck@6 {
-					regulator-compatible = "BUCK6";
-					regulator-name = "VCC_SUB_1.35V";
-					regulator-min-microvolt = <1350000>;
-					regulator-max-microvolt = <1350000>;
-					regulator-always-on;
-				};
-				buck7_reg: buck@7 {
-					regulator-compatible = "BUCK7";
-					regulator-name = "VCC_SUB_2.0V";
-					regulator-min-microvolt = <2000000>;
-					regulator-max-microvolt = <2000000>;
-					regulator-always-on;
-				};
-
-				buck8_reg: buck@8 {
-					regulator-compatible = "BUCK8";
-					regulator-name = "VMEM_VDDF_3.0V";
-					regulator-min-microvolt = <2850000>;
-					regulator-max-microvolt = <2850000>;
-					regulator-always-on;
-				};
-
-				buck9_reg: buck@9 {
-					regulator-compatible = "BUCK9";
-					regulator-name = "CAM_ISP_CORE_1.2V";
-					regulator-min-microvolt = <1000000>;
-					regulator-max-microvolt = <1200000>;
-				};
-			};
-		};
-	};
-
-	i2c_if_pmic: i2c@0 {
-		compatible = "i2c-gpio";
-		gpios = <&gpm2 0 0
-			 &gpm2 1 0
-			>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		max77693@66 {
-			compatible = "maxim,max77693";
-			interrupt-parent = <&gpx1>;
-			interrupts = <5 2>;
-			wakeup = <1>;
-			reg = <0x66>;
-
-			regulators {
-				esafeout@1 {
-					regulator-compatible = "ESAFEOUT1";
-					regulator-name = "ESAFEOUT1";
-					regulator-boot-on;
-				};
-				esafeout@2 {
-					regulator-compatible = "ESAFEOUT2";
-					regulator-name = "ESAFEOUT2";
-				};
-				charger@0 {
-					regulator-compatible = "CHARGER";
-					regulator-name = "CHARGER";
-					regulator-min-microamp = <60000>;
-					regulator-max-microamp = <2580000>;
-					regulator-boot-on;
-				};
-			};
-			muic_regs {
-				intmask@1 {
-					addr = <0x7>;
-					data = <0x9>;
-				};
-				intmask@2 {
-					addr = <0x8>;
-					data = <0x1>;
-				};
-			};
-		};
-	};
-
 	lcd_vdd_reg: voltage-regulator@1 {
 	        compatible = "regulator-fixed";
 		regulator-name = "LCD_VDD_5.0V";
@@ -578,34 +82,6 @@
 		enable-active-high;
 	};
 
-	cam_af_reg: voltage-regulator@2 {
-		compatible = "regulator-fixed";
-		regulator-name = "CAM_AF";
-		regulator-min-microvolt = <2800000>;
-		regulator-max-microvolt = <2800000>;
-		gpio = <&gpm0 4 0>;
-		enable-active-high;
-	};
-
-	cam_io_reg: voltage-regulator@3 {
-		compatible = "regulator-fixed";
-		regulator-name = "CAM_SENSOR_A";
-		regulator-min-microvolt = <2800000>;
-		regulator-max-microvolt = <2800000>;
-		gpio = <&gpm0 2 0>;
-		enable-active-high;
-	};
-
-	cam_isp_core_reg: voltage-regulator@4 {
-		compatible = "regulator-fixed";
-		regulator-name = "CAM_ISP_CORE_1.2V_EN";
-		regulator-min-microvolt = <1200000>;
-		regulator-max-microvolt = <1200000>;
-		gpio = <&gpm0 3 0>;
-		enable-active-high;
-		regulator-always-on;
-	};
-
 	fimd0_lcd: panel {
 		compatible = "samsung,s6d6aa1";
 		reset-gpio = <&gpf2 1 0>;
@@ -658,111 +134,10 @@
 		status = "okay";
 	};
 
-	vidi {
-		compatible = "samsung,exynos-drm-vidi";
-	};
-
-	sysmmu-fimd0 {
-		mmu-master = <&fimd>;
-	};
-
-	usbphy@125B0000 {
-		status = "okay";
-	};
-
-	hsotg@12480000 {
-		status = "okay";
-		vusb_d-supply = <&ldo15_reg>;
-		vusb_a-supply = <&ldo12_reg>;
-	};
-
-	spi_1: spi@13930000 {
-		pinctrl-names = "default";
-		pinctrl-0 = <&spi1_bus>;
-		status = "okay";
-
-		s5c73m3_spi: s5c73m3 {
-			compatible = "samsung,s5c73m3";
-			spi-max-frequency = <50000000>;
-			reg = <0>;
-			controller-data {
-				cs-gpio = <&gpb 5 0>;
-				samsung,spi-feedback-delay = <2>;
-			};
-		};
-	};
-
-	g2d@10800000 {
-		status = "okay";
-	};
-
 	rotator: rotator@12810000 {
 		status = "okay";
 	};
 
-	camera {
-		status = "okay";
-
-		pinctrl-names = "default";
-		pinctrl-0 = <&cam_port_a_clk_active>;
-
-		fimc_0: fimc@11800000 {
-			clock-frequency = <176000000>;
-			status = "okay";
-		};
-
-		fimc_1: fimc@11810000 {
-			clock-frequency = <176000000>;
-			status = "okay";
-		};
-
-		fimc_2: fimc@11820000 {
-			clock-frequency = <176000000>;
-			status = "okay";
-		};
-
-		fimc_3: fimc@11830000 {
-			clock-frequency = <176000000>;
-			status = "okay";
-		};
-
-		csis_0: csis@11880000 {
-			status = "okay";
-			vddcore-supply = <&ldo8_reg>;
-			vddio-supply = <&ldo10_reg>;
-			clock-frequency = <160000000>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			/* Camera C (3) MIPI CSI-2 (CSIS0) */
-			port@3 {
-				reg = <3>;
-				csis0_ep: endpoint {
-					remote-endpoint = <&s5c73m3_ep>;
-					data-lanes = <1 2 3 4>;
-					samsung,csis-hs-settle = <12>;
-				};
-			};
-		};
-	};
-
-	mfc: codec@13400000 {
-		status = "okay";
-
-		memport@0 {
-			memory-region = <&mfc_l_mem>;
-		};
-
-		memport@1 {
-			memory-region = <&mfc_r_mem>;
-		};
-	};
-
-	gpu@13000000 {
-		status = "okay";
-		vdd_g3d-supply = <&buck4_reg>;
-	};
-
 	cpufreq {
 		status = "okay";
 		overclocking = "okay";
@@ -779,36 +154,6 @@
 	};
 
 
-	modem_if {
-		compatible = "samsung,modem_if";
-		reset-req-gpio = <&gpm3 3 0>;
-		cp-on-gpio = <&gpl2 5 0>;
-		cp-reset-gpio = <&gpx3 2 0>;
-		pda-active-gpio = <&gpf1 6 0>;
-
-		phone-active-gpio = <&gpx1 6 0>;
-		cp-dump-int-gpio = <&gpx1 2 0>;
-
-		ap-dump-int-gpio = <0>;
-		flm_uart_sel = <0>;
-		cp_warm_reset = <0>;
-		sim_detect = <0>;
-
-		link-slavewake-gpio = <&gpx1 0 0>;
-		link-hostwake-gpio = <&gpx1 1 0>;
-		link-active-gpio = <&gpf1 1 0>;
-		link-enable-gpio = <0>;
-
-		modem-type = "xmm6262";
-		link-type = "hsic";
-		modem-net = "umts";
-		use-handover = <0>;
-
-		pinctrl-names = "default", "active";
-		pinctrl-0 = <&modem_state_off>;
-		pinctrl-1 = <&modem_state_active>;
-	};
-
 };
 
 &pinctrl_1 {
-- 
1.8.3.2

